DEV | GenerateData

sozkan4 opened the pull request:
Description 
Mock data models added for 1 admin and 2 normal users for 4 different spaces.

ceydaduzgec commented on core/management/generate_data.py:
```diff
@@ -0,0 +1,83 @@
+
+        print("Mock data created.")
+
+for space in spaces:
```
These should be inside of the handle function. Please correct this and push it again.
If you want to learn more about what we are doing here:
- https://simpleisbetterthancomplex.com/tutorial/2018/08/27/how-to-create-custom-django-management-commands.html
- https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/

ceydaduzgec commented on README.md:
```diff
@@ -122,22 +122,18 @@ This projects is for the SWE 574 Software Development Practice course of Boğazi
-
-See the [open issues](https://github.com/swe574-spring23/SWE574/issues) for a full list of proposed features (and known issues).
+To generate mock data for your local, you can use the following command, after the installation steps:
+  `python manage.py generate_data.py`
```
I realized I wrote the `.py` extra. It should be `python manage.py generate_data`

ceydaduzgec commented on core/management/generate_data.py:
```diff
@@ -0,0 +1,83 @@
+from django.core.management import BaseCommand
```
Also the file structure should be: `core/management/commands/generate_data.py` 
And `core`app should be included in `INSTALLED_APPS`Can you fix this as well?

ceydaduzgec commented on core/management/generate_data.py:
```diff
@@ -0,0 +1,83 @@
+
+for _ in range(4):  
+    user = User(
+        email=fake.email(),
```
Fake library is not included in imports. You can import them here or can use UserFactory to generate random users here.

ceydaduzgec commented on core/management/generate_data.py:
```diff
@@ -0,0 +1,83 @@
+        link=post_data["link"],
+    )
+    post.save()
+()
```
This () is extra please delete this and also please use the linter or we cannot merge this pull request without successfully passing the linter check.

ceydaduzgec reviewed (changes requested):
Great start but we need some little things that should be fixed. Let me know if something is unclear. Also can you fix the title of the pull request?

sozkan4 commented:
Hello Ceyda I fixed the code and test it working fine also fixed read me file. One concern here is maybe we need to update reuirements txt file for the faker? Thanks in advance.

ceydaduzgec commented on core/management/commands/generate_data.py:
```diff
@@ -0,0 +1,96 @@
+                space.save()
+
+    # Create posts with your specified data
+    posts_data = [
```
After this line the code is not indented as previously. Could you add on more tab to the code below please?

ceydaduzgec commented:
> Hello Ceyda I fixed the code and test it working fine also fixed read me file. One concern here is maybe we need to update reuirements txt file for the faker? Thanks in advance.

Hey, great work thanks!  We don't need to add faker since we were already using it other files. Only thing is this command is not working properly.  Could you find out the issue here please? If you want I can help you with that.

![image](https://user-images.githubusercontent.com/37382270/236000748-695467e1-53d6-4ef5-8773-e8b67a0b67ce.png)

sozkan4 commented:
@ceydaduzgec I think the problem is about model filtering and I think I fix it. Could you please give it a try?

ceydaduzgec commented:
@sozkan4 Looks amazing thanks a lot!
